transformer_engine/pytorch/transformer.py

```diff
@@ -158,6 +158,15 @@ class TransformerLayer(torch.nn.Module):
           The device on which the parameters of the model will allocated. It is the user's
           responsibility to ensure all parameters are moved to the GPU before running the
           forward pass.
+    hidden_states_format: str, default = 'sbhd'
```

Comment:
Hmmm, `hidden_states_format` is very verbose for an option that will be used pretty frequently I think. Can we have something shorter here? I wonder if just `format` or maybe `input_format` would be enough`....

Also, `sbhd` is not really correct for a format to TransformerLayer, since it implies 4D tensors, while TransformerLayer actually has 3D tensors as input.

And yet another thing - since the options here are very limited, you can list them in the documentation in this line, similar to what was done in `normalization` option: `{ 'sbhd', 'bshd' }`

Reply:
I think `input_format` (or maybe `input_layout`) sounds good. 
You're right about `sbhd` as it's only relevant for attention implementation. What if we make it `attn_input_format`? 

Otherwise, we'd either have to use [`sbd`, `bsd`] or [`batch_first`, `seq_first`] and translate them to `sbhd` or `bshd` inside the transformer layer. But then there will also be `td` format

Reply:
@ptrendx, what do you think?